src/Ocelot/Configuration/Creator/SecurityOptionsCreator.cs

```diff
@@ -5,35 +5,26 @@ namespace Ocelot.Configuration.Creator
 {
     public class SecurityOptionsCreator : ISecurityOptionsCreator
     {
-        public SecurityOptions Create(FileSecurityOptions securityOptions)
-        {
-            var ipAllowedList = new List<string>();
-            var ipBlockedList = new List<string>();
+        public SecurityOptions Create(FileSecurityOptions securityOptions, FileGlobalConfiguration globalConfiguration) 
+            => Create(securityOptions.IsFullFilled() ? securityOptions : globalConfiguration.SecurityOptions);
 
-            foreach (var allowed in securityOptions.IPAllowedList)
-            {
-                if (IPAddressRange.TryParse(allowed, out var allowedIpAddressRange))
-                {
-                    var allowedIps = allowedIpAddressRange.Select<IPAddress, string>(x => x.ToString());
-                    ipAllowedList.AddRange(allowedIps);
-                }
-            }
-
-            foreach (var blocked in securityOptions.IPBlockedList)
-            {
-                if (IPAddressRange.TryParse(blocked, out var blockedIpAddressRange))
-                {
-                    var blockedIps = blockedIpAddressRange.Select<IPAddress, string>(x => x.ToString());
-                    ipBlockedList.AddRange(blockedIps);
-                }
-            }
+        private static SecurityOptions Create(FileSecurityOptions securityOptions)
+        {
+            var ipAllowedList = SetIpAddressList(securityOptions.IPAllowedList);
+            var ipBlockedList = SetIpAddressList(securityOptions.IPBlockedList);
 
             if (securityOptions.ExcludeAllowedFromBlocked)
             {
-                ipBlockedList = ipBlockedList.Except(ipAllowedList).ToList();
+                ipBlockedList = ipBlockedList.Except(ipAllowedList).ToArray();
             }
 
             return new SecurityOptions(ipAllowedList, ipBlockedList);
         }
-    }
+
+        private static string[] SetIpAddressList(IList<string> ipValueList)
```

Comment:
Method is private! You should you native types for performance reasons.
Why do you use abstract `IList<string>` type to pass the `List<string>` argument aka `FileSecurityOptions` lists?
```suggestion
        private static string[] SetIpAddressList(List<string> ipValueList)
```
It is correct change if the method will be public, but private methods should have native type arguments.